src/key_pair.rs

```diff
@@ -55,19 +55,19 @@ impl KeyPair {
 	/// Parses the key pair from the DER format
 	///
 	/// Equivalent to using the [`TryFrom`] implementation.
-	pub fn from_der(der: &[u8]) -> Result<Self, RcgenError> {
-		Ok(der.try_into()?)
+	pub fn from_der(der: &[u8], rng: &dyn SecureRandom) -> Result<Self, RcgenError> {
```

Comment:
This extra required parameter unnecessarily complicates the api for most use cases.

I suggest keeping the original method signature, and providing an alternate method where the random number generator can be provided. Have the method with the original signature use a reasonable default.

Also, why is it `&dyn` ?

Reply:
> Also, why is it `&dyn` ?

Because it is a trait and therefore has to be passed by reference otherwise it is not `Sized` :)

> This extra required parameter unnecessarily complicates the api for most use cases.

Most usecases use `rcgen::generate_simple_self_signed` or `Certificate::from_params`, right? I'd argue that parsing keypairs etc is already low-level functionality for advanced usecases. If we duplicate all of these APIs with and without an RNG parameter, we end up with a lot of noise.

Reply:
Different question: is *ring* already part of the public API of the crate?

Reply:
I thought it was but upon checking again, it might not actually be!

Reply:
> I thought it was but upon checking again, it might not actually be!

That changes the game entirely. Any suggestions for how we can make `KeyPair`'s deterministic? I guess arguably, that should probably be done in a different PR.

Reply:
(Note that the use of ring error types in `rcgen::Error` also makes it public.)

I'm not saying making it a part of the public API is necessarily bad (not sure how intentional it was to keep it private), but just making sure we make this an explicit choice.

Reply:
> (Note that the use of ring error types in `rcgen::Error` also makes it public.)

Yeah good call. I'll make it private again.